src/Aspire.Hosting.Azure.Sql/AzureSqlServerResource.cs

```diff
@@ -108,11 +115,76 @@ public override ProvisionableResource AddAsExistingResource(AzureResourceInfrast
     public override void AddRoleAssignments(IAddRoleAssignmentsContext roleAssignmentContext)
     {
         var infra = roleAssignmentContext.Infrastructure;
-        var postgres = (SqlServer)AddAsExistingResource(infra);
+        var sqlserver = (SqlServer)AddAsExistingResource(infra);
 
-        var principalId = roleAssignmentContext.PrincipalId;
+        // The name of the user assigned identity for the container app
         var principalName = roleAssignmentContext.PrincipalName;
+        var principalId = roleAssignmentContext.PrincipalId;
 
-        AzureSqlExtensions.AddActiveDirectoryAdministrator(infra, postgres, principalId, principalName);
+        var sqlServerAdmin = UserAssignedIdentity.FromExisting("sqlServerAdmin");
+        sqlServerAdmin.Name = AdminName.AsProvisioningParameter(infra);
+        infra.Add(sqlServerAdmin);
+
+        foreach (var database in Databases.Keys)
+        {
+            // Create a unique resource identifier for the script resource
+            var hash = new XxHash3();
+            hash.Append(Encoding.UTF8.GetBytes(database));
+            hash.Append(Encoding.UTF8.GetBytes(infra.AspireResource.GetBicepIdentifier()));
+
+            var scriptResource = new SqlServerScriptProvisioningResource($"script_{Convert.ToHexString(hash.GetCurrentHash()).ToLowerInvariant()}");
+
+            scriptResource.Kind = "AzurePowerShell";
+            scriptResource.AZPowerShellVersion = "7.4";
+
+            // Run the script as the administrator
+            scriptResource.Identity.IdentityType = ArmDeploymentScriptManagedIdentityType.UserAssigned;
+            scriptResource.Identity.UserAssignedIdentities["${sqlServerAdmin.id}"] = new UserAssignedIdentityDetails();
```

Comment:
Look at:

https://github.com/dotnet/aspire/blob/935f06b15acaa8068375d5507423aacdaaf1de52/src/Aspire.Hosting.Azure.AppContainers/ContainerAppContext.cs#L78-L81

For the bizzare syntax 😄